app/views/events.py

```diff
@@ -145,42 +318,33 @@ def search_recipes(message) -> None:
             else:
                 recipe_ids.append(r.recipe_id)
 
-        recipe_phases: List[RecipePhase] = app_config.db.read_phases_from_recipes(recipe_ids)
-        
+        recipe_phases: List[RecipePhase] = app_config.db.read_phases_from_recipes(
+            recipe_ids
+        )
+
         recipes_json = [r.to_json() for r in matching_recipes]
         recipe_phases_json = [rp.to_json() for rp in recipe_phases]
 
         send_message_to_namespace_if_specified(
             socketio,
             message,
             "search_recipes_response",
-            {"succeeded": True, "recipes": recipes_json, "recipe_phases": recipe_phases_json},
+            {
+                "succeeded": True,
+                "recipes": recipes_json,
+                "recipe_phases": recipe_phases_json,
+            },
         )
-        
 
     @socketio.on("harvest_grow")
     def harvest_grow(message) -> None:
         print("message:", message)
-        if 'grow' not in message:
-            send_message_to_namespace_if_specified(
-                socketio,
-                message,
-                "harvest_grow_response",
-                {"succeeded": False, "reason": "Grow not included"},
-            )
+        if validate_harvest_grow(app_config, message) != (True, ""):
             return
```

Comment:
You need to return an error to the client, the client won't ever receive a response if you return without emitting a response event